diff -ru openttd-1.7.0/config.lib openttd-1.7.0-patched/config.lib
--- openttd-1.7.0/config.lib	2017-04-01 18:33:37.000000000 +0200
+++ openttd-1.7.0-patched/config.lib	2017-05-03 22:38:37.722468738 +0200
@@ -1479,12 +1479,6 @@
 			CFLAGS="-I/gg/os-include -noixemul -fstrict-aliasing -fexpensive-optimizations -mcpu=604 -fno-inline -mstring -mmultiple $CFLAGS"
 			LDFLAGS="$LDFLAGS -noixemul"
 		fi
-
-		if [ "$enable_profiling" = "0" ]; then
-			# -fomit-frame-pointer and -pg do not go well together (gcc errors they are incompatible)
-			CFLAGS="-fomit-frame-pointer $CFLAGS"
-		fi
-		CFLAGS="-O2 $CFLAGS"
 	else
 		OBJS_SUBDIR="debug"
 
@@ -1495,14 +1489,6 @@
 				CFLAGS="$CFLAGS -G0"
 			fi
 		fi
-		if [ $enable_debug -ge 2 ]; then
-			CFLAGS="$CFLAGS -fno-inline"
-		fi
-		if [ $enable_debug -ge 3 ]; then
-			CFLAGS="$CFLAGS -O0"
-		else
-			CFLAGS="$CFLAGS -O2"
-		fi
 	fi
 
 	if [ $enable_debug -le 2 ]; then
@@ -1527,11 +1513,6 @@
 		CFLAGS="$CFLAGS -fno-expensive-optimizations"
 	fi
 
-	if [ "$enable_profiling" != "0" ]; then
-		CFLAGS="$CFLAGS -pg"
-		LDFLAGS="$LDFLAGS -pg"
-	fi
-
 	if [ "$with_threads" = "0" ]; then
 		CFLAGS="$CFLAGS -DNO_THREADS"
 	fi
